tests/cpp/tests/test_nodes_indexing.cpp

```diff
@@ -1081,38 +1081,113 @@ TEST_CASE("AdvancedIndexingNode") {
                         verify_array_diff({}, {3, 5, 7, 2}, adv->diff(state));
                     }
 
-                    AND_WHEN("We mutate the main array") {
+                    AND_WHEN("We revert the indexing nodes") {
```

Comment:
Do we want to toss an `ArrayValidationNode` on `adv` while we're at it?

Reply:
Yeah, might as well. I can remove the `verify_array_diff()` calls as well.